src/core/os_macos.c

```diff
@@ -9,15 +9,19 @@
 #include <unistd.h>
 #include <time.h>
 #include <pwd.h>
+#import <AVFoundation/AVFoundation.h>
 
 #include "os_glfw.h"
 
-static uint64_t frequency;
+static struct {
+  uint64_t frequency;
+  fn_permission* onPermissionEvent;
```

Comment:
Maybe `os_destroy` should `memset(&state, 0, sizeof(state))` so the permission callback doesn't leak across restarts

Reply:
good call!